python-checks/src/test/resources/checks/fstringNestingLevel.py

```diff
@@ -0,0 +1,17 @@
+
+def non_compliant(hello, name):
+    my_string = f"{f"{f"{hello}"},"} {name}!" # Noncompliant {{Do not nest f-strings too deeply.}}
+    #           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
+
+    greeting = f"{f"{hello}"},"
+    my_string = f"hello:{greeting}, {name}, {f"end: { f"done" }"}!" # Noncompliant
+#               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 
+
+    my_string = f"hello { f"format specifier: {greeting :{F"1"}.{2}}"}" # Noncompliant
+#               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+def compliant(hello, name):
+    greeting = f"{f"{hello}"},"
+    my_string = f"{greeting} {name}!" # Compliant
+
+    my_string = f"{greeting} {name : { f"1" }.{2}}!" 
```

Comment:
It would maybe be good to add some more cases, that include more of the edge cases of the rule. Maybe have the use of escaped curly braces (`{{`), or play around with the number of quotes that the fstring implements.

Reply:
So the number of quotes is something I may not be able to test. In the sense that it should be an implementation detail of the StringElement, I should not have to test if the isInterpolated() method correctly reports if we are working with an interpolated string. This should be done at the level of the StringElement. For the escaped curly braces, it is a bit similar as I am only working with FromattedExpression I do not deal with the literals of the f-string. So the responsibility does not belong here in this case. If you want we can go into more details on this.